### ISSUE TITLE
Incomplete indexing of library

### ISSUE THREAD

**Issue by Javerre** [bpatrik/pigallery2#238]
I have 12,500 photos in my collection, organised into folders as:
  photos
    year
      year-month

However, when I try to view this in PiGallery2 it shows only a subset of the folders. For example, the top level set of folders is:
  2003 
  2004 
   ...
  2021

But the top level view is missing the 2003 folder. Drilling down, I similarly find sub-folders are missing all over the place.
I am using mysql as the database and the there tables are consistent with the rendered views (i.e. have similarly missing folder entries in the directory_entity table. 

All in all PiGallery2 only finds 2991 of my 12,500 media entries :(

Currently running 1.8.2 pulled from docker bpatrik/pigallery2:latest

**Comment by bpatrik** [comment]
Hi,

thanks, that sounds odd.
Can you check in the settings under `Folder indexing` -> `Exclude Folder List` and `Exclude File List` ?
It has some default values that makes some folder excluded from indexing.

If there isn't anything suspicious check logs if there is any error. In the `config.json` you can set the logging level to `silly` to print all logs.

**Comment by Javerre** [comment]
The exclusion lists look OK - I configured them to eliminate some unwanted stuff:
```
            "excludeFolderList": [
                ".Trash-1000",
                ".dtrash",
                "$RECYCLE.BIN",
                "@Recycle",
                ".@__thumb",
                ".DAV"
            ],
            "//[excludeFileList]": "Any folder that contains a file with this name will be excluded from indexing.",
            "excludeFileList": [
                ".picasa.ini",
                ".DS_Store"
            ]
```
Where are the logs? I'm running in Docker.

**Comment by bpatrik** [comment]
Please note that excludeFileList **excludes folders** that contains those
files, not the files itself. It might be that this is not what you want.
The app only indexes that it can handle (like photos and videos with
supported extension) no need to exclude other files.

You can find the logs with "docker logs <container name>
You can get the container name with "docker ps"




-- Sorry for being brief, sent from my phone.

Javerre ***@***.***> ezt írta (időpont: 2021. márc. 21., V
22:58):

**Comment by Javerre** [comment]
Ah.

Thanks, this is not a bug. Sorry to have wasted your time!

By the way, just wanted to congratulate you on a truly excellent implementation of picture zoom. I have tried *a lot* of web hosted galleries and yours is the first that I actually find useful! This is something I found supremely useful in Picasa and have missed ever since.

I have a couple of minor feature requests:
 - When zooming, keep the point under the cursor invariant rather than the centre of the view
 - When zoomed in, moving to next/previous pictures should keep the viewport unchanged (if possible)
 - Allow images to be deleted from the frontend

Otherwise, its perfect.
